subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -91,6 +91,42 @@ static int state_status_u8(struct bt_mesh_model *model,
 
 		bt_mesh_msg_ack_ctx_rx(&cli->ack_ctx);
 	}
+
+	if (cli->cb) {
```

Comment:
There is already choosing of separate handlers regarding opcodes. This is here: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/cfg_cli.c#L864
Please call appropriate callbacks in the corresponding status handler. No need for this huge `switch case` at all.